cmd/clusterctl/api/v1alpha3/labels.go

```diff
@@ -30,6 +30,9 @@ const (
 	// Example: resources shared between instances of the same provider:  CRDs,
 	// ValidatingWebhookConfiguration, MutatingWebhookConfiguration, and so on.
 	ClusterctlResourceLifecyleLabelName = "clusterctl.cluster.x-k8s.io/lifecycle"
+
+	// ClusterctlMoveLabelName can be set on objects that providers wish to move that are not part of a cluster
+	ClusterctlMoveLabelName = "clusterctl.cluster.x-k8s.io/move"
```

Comment:
Yeah, I plan to throw this label on all of my created resources for sure.

In regards to handling this at the CRD level, I think that could make some sense as a future improvement. As well as possibly following the ownerref chain on any resources of that CRD.

Reply:
Just a little more context. Clusterctl installs our "external" CRDs so the standard label gets applied automatically.

Also, I like the idea of automatically moving over all objects for a CRD that is marked for moving.

Reply:
To simplify and reduce a little the scope of this functionality, could we consider only applying a label to specific CRDs, or pre-defined custom objects (like secrets or configmaps, these should be already covered though with OwnerReferences). 

A label on a CRD can be defined such as: `clusterctl.cluster.x-k8s.io/manage-tree` (or similar, we can wordsmith as much as we want here). The functionality wouldn't be limited to move, but also other operations, for example delete. 

In the future it could be used to install custom generic providers, and define some new conventions. The reason I'm proposing to manage specific CRDs vs. moving ad-hoc single objects is to reduce potential issue that could arise from missing dependencies between a tree of objects, and reduce the scope of the feature to a well define set of actions that can be documented in a separate section.

Thoughts?

Reply:
I'm fine to configure move at CRD level as a future improvement

Instead, I think extending the current PR to "manage object tree", potentially outside the current scope of move, deserves more design